lib/src/core/verifier/sgx/policy.rs

```diff
@@ -0,0 +1,69 @@
+use super::*;
+use anyhow::Result;
+use std::path::PathBuf;
+use tokio::fs;
+
+impl Sgx {
+    pub fn opa_path(&self) -> Result<PathBuf> {
+        // opa/sgx/
+        Ok(self.workdir.join("opa").join("sgx"))
+    }
+
+    //Set default policy and reference file.
+    pub async fn default(&self) -> Result<()> {
+        let path = self.opa_path()?;
+        if !path.exists() {
+            fs::create_dir_all(path).await?;
+        }
+
+        let file = self.opa_policy_path()?;
+        if !file.exists() {
+            info!("{} isn't exist", file.to_str().unwrap());
+            let policy = r#"
+package policy
```

Comment:
Why not just include some default policies in the repo? Do we really need these functions to add the policies? It seems like this is something we only need to do once when we first install the AS.

Reply:
The background is this `AttesattionService` Crate has the features:
- It can be distributed as a single Crate.
- It should also support the potential multiple users scenario, its initial function's input includes a `user_id`, so different users will have their own policy.rego and reference files. And these files' path should also be different.
```
pub async fn new(path: &Path, user_id: String) -> Result<Self> {}
```

> Why not just include some default policies in the repo?

I think it's difficult to deploy these default policy and reference files via command directly (such as `make install`) in this multiple users scenario.

> Do we really need these functions to add the policies? It seems like this is something we only need to do once when we first install the AS.

I think they are needed, they will be invoked each time when any new `user_id` coresponding `AttesattionService` instance is created.

Reply:
Ok, I see where you're coming from.

If these are just default policies and reference data, couldn't they be shared between all users? We could have one directory that stores all the defaults (which are tracked in git as `.rego` file) and other directories could be created for each user.

I also wonder if the lib is the right place to handle manipulating files like this. It seems like maybe the app should be in charge of maintaining all per-user policy state. By the time the verifier is called all the policies should be in the right places.

Reply:
Yes, I am think you are right, it's reasonable to leave these files' creation to app program (such as `server`) totally. After the `AttestationService` instance is created by app, the app needs to inject these dependent policy files explicitly (Like the following codes).
```
let service = AttestationService::new(&Path::new("./"), user_id.clone()).await?;
service.policy_engine_set_file(<set policy file cmd>).await?;
service.policy_engine_set_file(<set reference data file cmd>).await?;
```

It can also help to keep the lib's logic as simple/clear as possible.

Reply:
Hi @fitzthum ,
I think this problem again and have the following new answers:

> If these are just default policies and reference data, couldn't they be shared between all users? We could have one directory that stores all the defaults (which are tracked in git as .rego file) and other directories could be created for each user.

Yes, these policy and reference files are shared by all users. And I will update the codes to make these files are tracked in git as `.rego`/`.json` files. 
And I think it should be better to put these files to their corresponding TEE folder. It makes the structure clear.

> I also wonder if the lib is the right place to handle manipulating files like this. It seems like maybe the app should be in charge of maintaining all per-user policy state. By the time the verifier is called all the policies should be in the right places.

I think it should be the AS Crate's responsibility to provide these default `policy.rego` and `reference_data.json` files. It's too hard for the Crate's new user to provide the `policy.rego` by themselves to make it work correctly. 
And the AS's default `policy.rego` is also a reference policy if the user want to update and enhance the policy later.

I also hunch that most AS Crate users won't care the `policy.rego` file, they will update the `reference_data.json` only.